src/ucs/vfs/base/vfs_obj.c

```diff
@@ -435,6 +463,32 @@ void ucs_vfs_obj_set_dirty(void *obj, ucs_vfs_refresh_cb_t refresh_cb)
     ucs_spin_unlock(&ucs_vfs_obj_context.lock);
 }
 
+char* ucs_vfs_obj_unique_dir_name(void *parent_obj, const char *rel_path, ...)
+{
+    ucs_vfs_node_t *parent_node;
+    va_list ap;
+    char rel_path_buf[PATH_MAX];
+    char abs_path_buf[PATH_MAX];
+
+    va_start(ap, rel_path);
+    vsnprintf(rel_path_buf, sizeof(rel_path_buf), rel_path, ap);
+    va_end(ap);
+
+    ucs_spin_lock(&ucs_vfs_obj_context.lock);
+
+    parent_node = ucs_vfs_node_get_by_obj(parent_obj);
+    if (parent_node == NULL) {
+        return NULL;
+    }
+
+    ucs_vfs_node_build_unique_path(parent_node, rel_path_buf, abs_path_buf,
+                                   sizeof(abs_path_buf));
+
+    ucs_spin_unlock(&ucs_vfs_obj_context.lock);
+
+    return ucs_strdup(ucs_basename(abs_path_buf), "ucs_vfs_obj_unique_name");
```

Comment:
why need to generate the whole path and then slice it? can we just generate the unique "leaf" directory name?